velox/expression/CastExpr.cpp

```diff
@@ -681,6 +705,8 @@ void CastExpr::applyPeeled(
     result = applyDecimal<int64_t>(rows, input, context, fromType, toType);
   } else if (toType->isLongDecimal()) {
     result = applyDecimal<int128_t>(rows, input, context, fromType, toType);
+  } else if (fromType->isBigint() && toType->isTimestamp()) {
+    result = castIntToTimestamp(rows, input, context);
```

Comment:
Maybe, we can support int8_t, int16_t, int32_t, int64_t with the same implementation like https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L639-L646.